pkg/crc/preflight/preflight_checks_common.go

```diff
@@ -158,7 +161,57 @@ func isDaemonRunningWithReleasedVersion() bool {
 	}
 	if ver.CrcVersion != version.GetCRCVersion() {
 		logging.Debugf("Daemon is running with %s version but binary version is %s", ver.CrcVersion, version.GetCRCVersion())
+		if err := killExistingDaemonProcess(); err != nil {
+			logging.Debugf("Failed to kill the existing crc daemon process: %v", err)
+		}
```

Comment:
This function is called `isDaemonRunningWithReleasedVersion`, it's not supposed to stop/kill the daemon as a side effect.

Reply:
The killing of the existing daemon process can be part of the fix function for this check?